src/System.Private.Xml/src/System/Xml/Core/XmlTextWriter.cs

```diff
@@ -338,15 +343,45 @@ public int Indentation
             {
                 if (value < 0)
                     throw new ArgumentException(SR.Xml_InvalidIndentation);
-                _indentation = value;
+                if(_indentation != value)
+                {
+                    _indentation = value;
+                    int arraySize = _indentation * IndentArrayLength;
+                    if (arraySize > _indentChars.Length)
+                    {
+                        char indentChar = _indentChars[0];
+                        _indentChars = new char[arraySize];
```

Comment:
I think we can keep the set method of this property as-is. We don't need to create a new array when `arraySize > _identChars.Length` as we can always use the array of length 64 and let [line#1444](https://github.com/dotnet/corefx/pull/21155/files#diff-b8ae98c49374192a92d1fd1f6985e764R1444) take care of the rest.